pkg/flow/internal/controller/component_references.go

```diff
@@ -42,8 +43,13 @@ func ComponentReferences(parent *vm.Scope, cn dag.Node, g *dag.Graph) ([]Referen
 
 	refs := make([]Reference, 0, len(traversals))
 	for _, t := range traversals {
-		// Determine if a reference refers to something existing.
-		if _, ok := parent.Lookup(t[0].Name); ok {
+		// We use an empty scope to determine if a reference refers to something in
+		// the stdlib, since vm.Scope.Lookup will search the scope tree + the
+		// stdlib.
+		//
+		// Any call to an stdlib function is ignored.
+		emptyScope := &vm.Scope{Variables: stdlib.Identifiers}
```

Comment:
This is no longer an empty scope, and the comment explaining how Lookup works is no longer relevant, since we're shadowing the default scope.